### DISCUSSION TITLE
Number of frames in CSV file doesn’t match that in the video file.

### DISCUSSION THREAD

**Discussion by KangByungwoo** [bonsai-rx/bonsai#1486] [Q&A]
I am using CSVWriter and VideoWriter nodes to create a csv file that has metadata for each frame of the video file. But, the number of rows in the CSV file doesn’t always match the number of frames in the video file as inferred by cv2 module in Python. Does anyone know why this happens and how to fix this? Thank you!

**Comment by bruno-f-cruz** [comment]
Please include the workflow.

**Reply by KangByungwoo** [reply]
![image](https://github.com/bonsai-rx/bonsai/assets/26239966/207a7240-f825-444e-a1e7-47e0233f28db)

**Reply by KangByungwoo** [reply]
I am using Flea3 FL3-U3-13Y3M cameras. And FlyCapture nodes are PointGrey2.FlyCatpureDataFrame nodes.

**Comment by bruno-f-cruz** [comment]
Can you try to move the `CsvWriter` between the `Timestamp` and `Value.Image`? You can use the `Selector` property to only select the what you want to save. I am curious to know if this would fix the problem.

**Reply by KangByungwoo** [reply]
How can I move the CsvWriter between the Timestamp and Value.Image?

**Reply by bruno-f-cruz** [reply]
`Enter` or `Shift+Enter` when selecting an entry in the toolbox.

**Reply by KangByungwoo** [reply]
Do you mean a workflow like this? 
![image](https://github.com/bonsai-rx/bonsai/assets/26239966/ed07dabc-806e-4acb-999a-8c315cae9e15)

**Reply by KangByungwoo** [reply]
So far, this seems to fix the problem.

**Comment by bruno-f-cruz** [comment]
Yes
